tests/integration/test_auth_proxy.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for jenkins-k8s-operator with auth_proxy."""
+
+# pylint: disable=unused-argument
+
+import re
+import time
+
+import pytest
+import requests
+from juju.application import Application
+from juju.model import Model
+from playwright.async_api import expect
+from playwright.async_api._generated import Page
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.asyncio
+async def test_auth_proxy_integration_returns_not_authorized(
+    model: Model,
+    application: Application,
+    oathkeeper_related: Application,
+) -> None:
+    """
+    arrange: deploy the Jenkins charm and establish auth_proxy relations.
+    act: send a request Jenkins.
+    assert: a 401 is returned.
+    """
+    status = await model.get_status()
+    address = status["applications"]["traefik-public"]["public-address"]
+    response = requests.get(  # nosec
```

Comment:
What is the nosec here please?

Reply:
Added an inline comment with the explanation